src/index.ts

```diff
@@ -110,3 +115,9 @@ export async function cli(args: string[]) {
   console.log(`Unrecognized command: ${cmd}`);
   process.exit(1);
 }
+
+/** Snowpack Build Plugin type */
+export type SnowpackPlugin<T = any> = (
```

Comment:
Thoughts on changing `T` to `PluginOptions`?

Reply:
Also, thoughts on a stricter type fallback for the generic? Should it be an object?

Reply:
+1 on PluginOptions, given that this only exists for users to better understand the plugin. 
+1 on the default matching the default when none is given, but can't remember if that's empty object, undefined, etc.

Reply:
Can a plugin be defined without the function just passing the object?

Reply:
Currently we don’t allow initializing a plugin without calling a function, but we could change [this line](https://github.com/pikapkg/snowpack/blob/master/src/config.ts#L350) to make that possible if desired (I’m a soft -1 against having multiple ways to do things, but don’t feel super strongly if this is a pattern people like)

Reply:
> soft -1 against having multiple ways to do things

I generally agree. Wasn't sure if it would be useful in the case of your example where they're not utilizing the factory portion and really just providing an object.

I'm fine with whichever decision you decide.

Reply:
Sure. I think since 2.7 has been delayed long enough, and all tests and templates are passing, that I’d like to keep it function-only for now. But it’s something we can easily add as a follow-up if desired, if that’s all right.